baseapp/abci.go

```diff
@@ -340,6 +340,23 @@ func (app *BaseApp) Commit() (res abci.ResponseCommit) {
 		halt = true
 	}
 
+	// each listener has an internal wait threshold after which it sends `false` to the ListenSuccess() channel
+	// but the BaseApp also imposes a global wait limit
+	maxWait := time.NewTicker(app.globalWaitLimit * time.Second)
```

Comment:
you should `defer` this right after the ticker is created.
```suggestion
	maxWait := time.NewTicker(app.globalWaitLimit * time.Second)
	defer maxWait.Stop()
```